main.go

```diff
@@ -246,13 +322,28 @@ func (m *Miner) hashratePrinter() {
 }
 
 // resultLoop takes in the result and passes to the proper channels for receiving.
-func (m *Miner) resultLoop() error {
+func (m *Miner) resultLoop(config util.Config) {
 	for {
 		select {
 		case header := <-m.resultCh:
 			order, err := m.GetDifficultyOrder(header)
 			if err != nil {
+				// Ignore this block and go back to waiting for another one.
 				log.Println("Block mined has an invalid order")
+				continue
+			}
+			if !config.Proxy {
```

Comment:
can we make proxy as the flag in Miner struct? then no need to pass in the config to this function

Reply:
Good idea